### ISSUE TITLE
Implements arg order is backwards

### ISSUE THREAD

**Issue by willfaught** [stretchr/testify#146]
Most non-expected-actual functions, like EqualError and Len, put the object to be examined first and the expected value second, but Implements does it in reverse order: type first, then object. This makes it hard to remember.

**Comment by matryer** [comment]
Please fix this and submit a PR. It would be great if it were consistent.

**Comment by phemmer** [comment]
I would argue that `Implements` is correct, and `Len` is backwards.
In most of the functions, the thing being tested against is the second argument, and the thing being tested is the third (and latter) argument(s). Thus in `Len`, the expected value should be the second argument.

Functions I think need to be fixed: `Contains`, `Len`, `HTTPBodyContains`, `HTTPBodyNotContains`, `NotContains`

**Comment by willfaught** [comment]
Agreed.

**Comment by phemmer** [comment]
Looks like another backwards method has been added since this issue was created: `EqualError`

**Comment by mvdkleijn** [comment]
I believe this issue and related PR can move forward again, but we need to take a look at #399 which makes a fair point.

**Comment by willfaught** [comment]
I agree with https://github.com/stretchr/testify/issues/399, actually. [Go has a strong preference for the actual-then-expected order of comparison values](https://github.com/golang/go/wiki/CodeReviewComments#useful-test-failures). This ticket is just about the order—whatever it may be—being consistent among the funcs.

**Comment by willfaught** [comment]
Closing due to no activity.

**Comment by zzh8829** [comment]
request to reopen this for v2, the v1 parameter ordering is very counter intuitive

**Comment by tonyhb** [comment]
+1 - `Len` is inconsistent.  The asserted value comes last, vs EqualValues which comes first.  In general, a bunch of these are inconsistent and it's pretty frustrating.

**Comment by brackendawson** [comment]
As this is seems to be the oldest one, and it's already in the v2.0.0 milestone, sure. If v2 ever happens the expected/actual order will be made consistent. Most likely using the order from `assert.Equal` as it's the most ubiquitous.

**Comment by HeCorr** [comment]
In regards to the naming convention, I vote for renaming the sides to "left" and "right" like they are called in Rust, eliminating the discrepancy between all the functions and making the sides unambiguous no matter which standard you're used to (expected-actual/actual-expected).

Thoughts? @willfaught @brackendawson

**Comment by willfaught** [comment]
That’s what I’ve done in my own test helper. Given the age of this issue, I doubt this will ever happen to testify. Better to make your own, better library.

**Comment by Antonboom** [comment]
Covered by [testifylint#expected-actual](https://github.com/Antonboom/testifylint?tab=readme-ov-file#expected-actual)

**Comment by brackendawson** [comment]
Yes, left v right is interesting. But we're not going to change the order/name of any assertions in v1, even to make them consistent in the module. At this time I don't anticipate a v2.
